staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -58,6 +58,26 @@ func (obj *Unstructured) IsList() bool {
 	_, ok = field.([]interface{})
 	return ok
 }
+func (obj *Unstructured) ToList() (*UnstructuredList, error) {
+	if !obj.IsList() {
+		// return an empty list back
+		return &UnstructuredList{Object: obj.Object}, nil
```

Comment:
this doesn't seem correct... shouldn't this either return an error or put the object in Items?

Reply:
This happens on partial input.  Logically you may have no items. In which case islist says false, but you know it's a list and needs to be empty.